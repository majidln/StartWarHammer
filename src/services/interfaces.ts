export interface World {
  id: String;
  name: String;
}

export interface Film {
  id: String;
  title: String;
}

export interface Entity {
  id: String;
  name: String;
  __typename: String;
}

export interface Starship extends Entity {
  crew?: Number;
  length?: Number;
  films?: Array<Film>;
}
export interface Person extends Entity {
  height?: Number;
  mass?: Number;
  homeworld?: World;
  films?: Array<Film>;
}
export interface Planet extends Entity {
  population?: Number;
  gravity?: String;
}

export interface ResponsePage {
  starships: Array<Starship>;
  persons: Array<Person>;
  planets: Array<Planet>;
}

export interface QueryResponse {
  data: ResponsePage | undefined;
  loading: Boolean;
  error: Object | undefined;
  refetch: Function;
}
