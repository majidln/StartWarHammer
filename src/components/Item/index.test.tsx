import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, waitFor, fireEvent} from '@testing-library/react-native';
import {Item} from './index';

import {Entity} from '@services/interfaces';

const itemStarship: Entity = {
  id: 'cj0nwtqudq53b0114s5id3anc',
  name: 'A-wing',
  __typename: 'Starship',
};

const itemPerson: Entity = {
  id: 'cj0nwtqudq53b0114s5id3anc',
  name: 'A-wing',
  __typename: 'Person',
};

it('Item render in list correctly', async () => {
  const {findByTestId} = render(<Item item={itemStarship} isDetail={false} />);

  await waitFor(() => findByTestId('itemID'));
});

it('Item button press in list working correctly', async () => {
  const {findByTestId} = render(<Item item={itemPerson} isDetail={false} />);

  const button = await findByTestId('itemBtn');
  fireEvent.press(button);
});

it('Item button press in detail working correctly', async () => {
  const {findByTestId} = render(<Item item={itemPerson} isDetail={true} />);

  const button = await findByTestId('itemBtn');
  fireEvent.press(button);
});
