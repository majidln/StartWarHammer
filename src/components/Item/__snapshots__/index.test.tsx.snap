// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "margin": 8,
      "padding": 10,
      "paddingBottom": 30,
    }
  }
  testID="itemID"
>
  <Text
    style={
      Object {
        "fontFamily": "StarJedi",
        "fontSize": 20,
        "fontWeight": "700",
      }
    }
  >
    A-wing
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "400",
      }
    }
  >
    Type: 
    Starship
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "opacity": 1,
        "padding": 8,
        "position": "absolute",
        "right": 0,
      }
    }
    testID="itemBtn"
  >
    <Text
      style={
        Object {
          "color": "red",
        }
      }
    >
      Show Detail
    </Text>
  </View>
</View>
`;
